server/src/main/java/com/cloud/network/IpAddressManagerImpl.java

```diff
@@ -732,8 +732,9 @@ public boolean disassociatePublicIpAddress(long addrId, long userId, Account cal
         PublicIpQuarantine publicIpQuarantine = null;
         // Cleanup all ip address resources - PF/LB/Static nat rules
         if (!cleanupIpResources(addrId, userId, caller)) {
-            success = false;
-            s_logger.warn("Failed to release resources for ip address id=" + addrId);
+            String msg = String.format("Failed to release resources for ip address id=%s", addrId);
+            s_logger.error(msg);
+            throw new CloudRuntimeException(msg);
```

Comment:
does this block any other subsequent calls?

Reply:
it does, but I think the process should be stopped here if the error happens.

Reply:
I think this is very much unreadible code:
- in a method named `disassociatePublicIpAddress(..)`
- after a call to `cleanupIpResources(..)`
- folows a call to `markIpAsUnavailable(..)` 
at the very least the last one must be renamed. I ropose `markIpForReleasing(..)`. Looking in the code, that is what it actually does.

I agree that some action other than marking for releasing should follow a failure to clean up. Let's investigate the callers and see what can be done.

Reply:
Hi @weizhouapache should these be addressed here, or later in separate PR?

Reply:
@sureshanaparti @DaanHoogland 
renamed the last method from `markAsUnavailable` to `markAsReleasing`